### ISSUE TITLE
Quick switch between shapes (rect <-> circle, etc.)

### ISSUE THREAD

**Issue by Mrazator** [excalidraw/excalidraw#8520]
Add the ability to switch between different shapes quickly, keeping the original shape style and bound text size.

The current way of doing this includes a lot of manual steps:

- turn on snapping and create a new shape of a similar size
- copy the styles from the original shape and paste the style into the new shape
- unbind text from the original shape and bind it to the new shape (usually the size of the bound text is different than the built-in size)

Instead, changing from rect <-> circle/diamond should be doable with a single click of a button, for all the selected elements.

**Comment by imran1khan** [comment]
so basically you want to copy past the text  when we give styles to new shapes , currect me if i am wrong? because when we copy the style from one shape and past it to another the binded text does not apper right?

**Comment by Mrazator** [comment]
Not exactly, as copying the styles (with or without the bound text) implies that you are going to apply the styles to another element, that you manually created.

Instead what I am suggesting is to have the possibility to change the shape/type of all the existing element/s without creating a new one. It should also preserve all the bindings, position, z-index, text wrapping, etc. Given that we could come up with some simple & reasonable UX.

**Comment by imran1khan** [comment]
hi @Mrazator  let me know your, is it fine?

**Comment by its-kunal** [comment]
Hey @Mrazator, can i start working on this issue.

**Comment by mtolmacs** [comment]
> Hey @Mrazator, can i start working on this issue.

@its-kunal There is a PR waiting to be merged delivering this functionality: #8533

You can check it out and test it if you'd like to help.

**Comment by Mrazator** [comment]
Implemented in #9270

**Comment by Achuan-2** [comment]
Looking forward to the release
